vsintegration/tests/unittests/IndentationServiceTests.fs

```diff
@@ -37,36 +39,133 @@ namespace testspace
         static member testmember = 1
 
 "
+
+    static let autoIndentTemplate = "
+let plus x y =
+    x + y
+
+let mutable x = 0
+x <-
+    10 * 2
+
+match Some 10 with
+| None -> 0
+| Some x ->
+    x + 1
+
+try
+    failwith \"fail\"
+with
+    | :? System.Exception -> \"error\"
```

Comment:
We aren't testing whether this line gets indented.

Also these test cases would be far easier to read if instead of specifying which lines we expect indentation in a separate array, we specified them inline in the template.

For example, these few lines would become:

```f#
try
    failwith \"fail\" // $Indent: 4$
with
    | :? System.Exception -> \"error\" // $Indent: 4$
```

You could then parse the template like:

```f#
let indentComment = System.Text.RegularExpressions.Regex(@"\/\/\s*\$\s*Indent:\s*(\d+)\s*\$")

template.Split [|'\n'|]
|> Seq.map (fun s -> s.Trim())
|> Seq.indexed
|> Seq.choose (fun (line, text) ->
    let m = indentComment.Match text
    if m.Success then Some (line, System.Convert.ToInt32 m.Groups.[1].Value)
    else None
)
```

This will return you a list of `lineNumber * indentSize`.